device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceClient.java

```diff
@@ -247,7 +292,24 @@ public DeviceClient(String connString, IotHubClientProtocol protocol, SSLContext
      */
     public static DeviceClient createFromSecurityProvider(String uri, String deviceId, SecurityProvider securityProvider, IotHubClientProtocol protocol) throws URISyntaxException, IOException
     {
-        return new DeviceClient(uri, deviceId, securityProvider, protocol);
+        return new DeviceClient(uri, deviceId, securityProvider, protocol, null);
+    }
+
+    /**
+     * Creates a device client that uses the provided security provider for authentication.
+     *
+     * @param uri The connection string for iot hub to connect to (format: "yourHubName.azure-devices.net")
```

Comment:
URI is not connection string. Isn't it hostname only?

Reply:
it is hostname. will modify this comment